internal/routes/apiserver.go

```diff
@@ -49,6 +49,23 @@ func APIServer(
 ) (http.Handler, func(), error) {
 	closer := func() {}
 
+	// Configure system token signing keys. This is done here to ensure the
+	// bootstrapping is shared with the test harnesses, even though it's not
+	// strictly "route" setup.
+	if _, err := db.ActiveTokenSigningKey(); err != nil {
+		if !database.IsNotFound(err) {
+			return nil, closer, fmt.Errorf("failed to lookup active token signing key: %w", err)
+		}
+
+		tokenSignerTyp, ok := tokenSigner.(keys.SigningKeyManager)
+		if !ok {
+			return nil, closer, fmt.Errorf("token signing key manager is not a signing key manager (is %T)", tokenSigner)
+		}
+		if _, err := db.RotateTokenSigningKey(ctx, tokenSignerTyp, cfg.TokenSigning.ParentKeyName(), database.System); err != nil {
```

Comment:
I think RotateTokenSigningKey needs some kind of locking mechanism (distributed).

It's possible that multiple container instances would try and do this at the same time.

Reply:
Chatted about this sync. We're going to allow the rotation service to bootstrap the initial key.